apps/public/public.go

```diff
@@ -297,8 +298,8 @@ func (pca Client) AcquireTokenSilent(ctx context.Context, scopes []string, opts
 	if err := options.ApplyOptions(&o, opts); err != nil {
 		return AuthResult{}, err
 	}
-	// a home account ID is required to find user tokens in the cache
-	if o.account.HomeAccountID == "" {
+	// an account is required to find user tokens in the cache
+	if reflect.ValueOf(o.account).IsZero() {
 		return AuthResult{}, errNoAccount
```

Comment:
@chlowell would you guys consider exposing this errNoAccount as ErrNoAccount so us devs can use errors.Is()?  I'm doing a string comparison currently and it feels messy.

https://github.com/bdwyertech/go-az/blob/529fe3c0feca36fc1cab1e74774222f589c9ecc3/pkg/az/auth.go#L117